enterprise/suppress-user/handler.go

```diff
@@ -0,0 +1,57 @@
+package suppression
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/samber/lo"
+)
+
+// newHandler creates a new handler for the suppression feature
+func newHandler(ctx context.Context, r Repository, log logger.Logger) *handler {
+	h := &handler{
+		r:   r,
+		log: log,
+	}
+	// we don't want to flood logs if, e.g. the suppression repository is restoring,
+	// so we are debouncing the logging
+	h.errLog.debounceLog, h.errLog.cancel = lo.NewDebounce(1*time.Second, func() {
+		h.errLog.errMu.Lock()
+		defer h.errLog.errMu.Unlock()
+		if h.errLog.err != nil {
+			h.log.Warn(h.errLog.err.Error())
+		}
+	})
+	go func() {
+		<-ctx.Done()
+		h.errLog.cancel()
+	}()
+	return h
+}
+
+// handler is a handle to this object
+type handler struct {
+	log    logger.Logger
+	r      Repository
+	errLog struct {
+		debounceLog func() // logs suppression failures with a debounce, once every 1 second
+		cancel      func() // cancels the debounce timer
+		errMu       sync.Mutex
+		err         error // the last error
+	}
```

Comment:
I am having hard time understanding the exact use case of this, I would really appreciate some explanation here. 😅

Reply:
the comment in line 19 tries to explain the reason we are using it
```
// we don't want to flood logs if, e.g. the suppression repository is restoring,
// so we are debouncing the logging
```